src/mgmt/srv.c

```diff
@@ -409,26 +427,31 @@ process_createpool_request(Drpc__Call *drpc_req,
 	mgmt__create_pool_req__free_unpacked(daos_req, NULL);
 	if (svc)
 		d_rank_list_free(svc);
-	if (rc != 0)
-		daos_resp->status = MGMT__DAOS_REQUEST_STATUS__ERR_UNKNOWN;
+	if (rc != 0) {
+		daos_resp->status->state =
+			MGMT__DAOS_REQUEST_STATUS__STATE__ERR_DAOS;
+		daos_resp->status->der = rc;
+	}
 }
 
 static void
-process_destroypool_request(Drpc__Call *drpc_req, Mgmt__DaosResp *daos_resp)
+process_destroypool_request(Drpc__Call *drpc_req,
+			    Mgmt__DestroyPoolResp *daos_resp)
```

Comment:
Just a nit, but the name "daos_resp" is a little weird now that the type isn't DaosResp.